models/weecology-esss.R

```diff
@@ -0,0 +1,155 @@
+
+      for(i in 1:nmoons){
+        if(length(which(abundances$newmoonnumber == moons[i])) > 0){
+          abundst <- abundances[which(abundances$newmoonnumber == moons[i]),
+                                which(colnames(abundances) %in% species)]
+          abunds[i, ] <- as.numeric(abundst)
+        }
+      }
+
+      interpolated_abunds <- abunds
+
+      for(j in 1:nspecies){
+        interpolated_abunds[ , j] <- round(na.interp(abunds[, j]))
+      }
+
+      interpolated_total <- apply(interpolated_abunds, 1, sum)
+
+    # fit the ets model and forecast with it
+
+      etsm <- ets(interpolated_total)
+      etsmf <- forecast(etsm, h = num_forecast_months,
+                          level = CI_level, 
+                          allow.multiplicative.trend = TRUE)
+
+    # prep the forecast data tabe
+
+      fdt <- data.frame(date = forecast_date, 
+                        forecastmonth = forecast_months,
+                        forecastyear = forecast_years, 
+                        newmoonnumber = forecast_newmoons,
+                        currency = "abundance",
+                        model = "Weecology-ESSS", 
+                        level = level, 
+                        species = "total", estimate = etsmf$mean,
+                        LowerPI = etsmf$lower[,
+                                      which(etsmf$level == CI_level*100)], 
+                        UpperPI = etsmf$upper[,
+                                      which(etsmf$level == CI_level*100)])
+       fdt[sapply(fdt, is.ts)] <- lapply(fdt[sapply(fdt, is.ts)], unclass)
+  
+       # Include columns describing the data used in the forecast
+
+         fdt$fit_start_newmoon <- min(abundances$newmoonnumber)
+         fdt$fit_end_newmoon <- max(abundances$newmoonnumber)
+         fdt$initial_newmoon <- max(abundances$newmoonnumber)
+  
+    # prep the aic data tabe
+      
+      aic <- data.frame(date = as.Date(forecast_date), 
+                        currency = 'abundance', 
+                        model = 'Weecology-ESSS', 
+                        level = level, species = 'total', 
+                        aic = as.numeric(etsmf$model$aic), 
+                        fit_start_newmoon = min(abundances$newmoonnumber),
+                        fit_end_newmoon = max(abundances$newmoonnumber), 
+                        initial_newmoon = max(abundances$newmoonnumber))
+
+    # return the output
+
+      return(list(fdt, aic))
+  }
+
+
+
+
+# Run model on all plots and just controls
+
+  # Get data
+
+    all <- read.csv("data/rodent_all.csv")
+    controls <- read.csv("data/rodent_controls.csv")
+    model_metadata <- yaml.load_file("data/model_metadata.yaml")
+    forecast_date <- as.Date(model_metadata$forecast_date)
+    filename_suffix <- model_metadata$filename_suffix
+    forecast_months <- model_metadata$forecast_months
+    forecast_years <- model_metadata$forecast_years
+    forecast_newmoons <- model_metadata$forecast_newmoons
+
+  # Forecast All plots
```

Comment:
The content of this content should be captured in the file name and the comment removed. If the current `allresults` doesn't quite do that for you then `all_forecasts` or `forecasts_all` or something.

You can also make things more explicit by use verbs for function calls. E.g.,

```
all_forecasts <- forecast_ess(...
```

or

```
all_forecasts <- get_ess_forecasts(...
```